tensorflow/python/data/experimental/ops/optimization_options.py

```diff
@@ -126,6 +126,13 @@ class OptimizationOptions(options.OptionsBase):
       "Whether to fuse filter dataset that predicts random_uniform < rate into "
       "a sampling dataset. If None, defaults to False.")
 
+  hoist_data_discarding_ops = options.create_option(
+      name="hoist_data_discarding_ops",
+      ty=bool,
+      docstring=
+      "Whether to hoist ops that will discard data (such as skip, take, shard)"
+      "out of map transformations. If None, defaults to False.")
```

Comment:
"out of map transformations" is misleading (and incomplete)

Reply:
modified.